tests/test_vault.py

```diff
@@ -0,0 +1,185 @@
+from ocw.lib.vault import Vault, AzureCredential, EC2Credential, GCECredential
+from datetime import datetime
+from datetime import timedelta
+import pytest
+import requests
+import webui
+import json
+import string
+import random
+import base64
+
+
+# Global test data
+namespace = 'test-qa'
+host = 'http://foo.bar'
+leases = list()
+
+
+def random_str(stringLength):
+    letters = string.ascii_letters
+    return ''.join(random.choice(letters) for i in range(stringLength))
+
+
+class MockResponse:
+    def __init__(self, json_response={}, status_code=200):
+        self.json_response = json_response
+        self.content = json.dumps(json_response)
+        self.status_code = status_code
+
+    def json(self):
+        return self.json_response
+
+
+def mock_post(url, **kwargs):
+    if (url.startswith('http://foo.bar/v1/auth/userpass/login/')):
+        username = url[38:]
+        password = kwargs['json']['password']
+        if (username == 'devel' and password == 'sag_ich_nicht'):
+            return MockResponse({'auth': {'client_token': 'TOTAL_SECRET_TOKEN', 'lease_duration': 60*15}})
+        return MockResponse({'errors': ['Unknown user']}, status_code=401)
+
+    if (url.startswith('{}/v1/sys/leases/revoke'.format(host))):
+        lease_id = kwargs['json']['lease_id']
+        try:
```

Comment:
hm try/catch inside Mock object ? we need to discuss this ...

Reply:
you could also just check, if this value exists in that array...